lib/nexpose/node.rb

```diff
@@ -83,22 +83,28 @@ def method_missing(method, *args)
       # Finally the enumerations: names
       if method_name == 'names'
         @xml.xpath('./names/name').collect(&:text)
-
       elsif ['fingerprints', 'software'].include?(method_name)
-
-        xpath_selector = {
-          'fingerprints' => './fingerprints/os',
-          'software' => './software/fingerprint'
-        }[method_name]
-
-        xml_os = @xml.at_xpath(xpath_selector)
-        return '' if xml_os.nil?
-
-        xml_os.attributes['product'].value
+        get_fingerprint_product(method_name)
       else
         # nothing found, the tag is valid but not present in this ReportItem
         return nil
       end
     end
+
+    private
+
+    # returns the first 'product' value from the <fingerprints> or <software> element
+    def get_fingerprint_product(method_name)
+      xpath_selector = {
+        'fingerprints' => './fingerprints/os',
+        'software' => './software/fingerprint'
+      }[method_name]
+
+      xml_fingerprint = @xml.xpath(xpath_selector)
+      return '' if xml_fingerprint.nil?
+
+      product = xml_fingerprint.find { |fingerprint| fingerprint['product'] }
```

Comment:
this code isn't very obvious at first glance, if `xml_finterprints` is a collection, is `#find` like a filter that returns on first success? Seems like a nokogiri idiom rather than a ruby one.

Also, is |fingerprint| the best loop variable name? Could it be `element` or `node`? At least to make it less confusing?